javascript/javascript1/week3/main.js

```diff
@@ -0,0 +1,274 @@
+
+const notes = [];
+var i = 0;
+function saveNote(content, id) {
+  notes.push({});
+  for (; i < notes.length; i++) {
+    notes[i].content = content;
+    notes[i].id = id;
+  }
+}
+
+saveNote("Pick up groceries", 1);
+saveNote("Do laundry", 2);
+saveNote("Run dishwasher", 3);
+saveNote("Cook meal", 4);
+
+console.log(notes);
+
+function getNote(id) {
+  if (typeof id == "number") {
+    for (i = 0; i < notes.length; i++) {
+      if (notes[i].id === id) {
+        return notes[i];
+      } else {
+        const error1 = "Error: This id does not exist.";
+        return error1;
+      }
+    }
+  } else {
+    const error2 = "Error: Please enter a number.";
+    return error2;
+  }
+}
+
+const firstNote = getNote(6);
+console.log(firstNote);
+
+function logOutNotesFormatted() {
+  for (var i = 0; i < notes.length; i++) {
+    return (
+      "The note with id: " +
+      notes[i].id +
+      ", has the following note text: " +
+      notes[i].content
+    );
+  }
+}
+
+console.log(logOutNotesFormatted());
+
+/*Unique feature: to remove a note once done 
+using content and returns newNotes with 
+remaining contents and id*/
+
+const newNotes = [...notes]; // make a clone of notes to remove content and to keep the original version of notes
+function removeNote(content) {
+  for (var i = 0; i < newNotes.length; i++) {
+    if (newNotes[i].content === content) {
+      newNotes.splice(i, 1);
+      i--;
+    }
+  }
+  return newNotes;
+}
+
+console.log(removeNote("Do laundry"));
+
+//CactusIO-interactive (Smart phone usage app) optional
+const activities = [];
+var i = 0;
+function addActivity(data, activity, duration) {
+  if (data === undefined || activity === undefined || duration === undefined) {
+    return "Add some activities before calling showStatus";
+  }
+  activities.push({});
+  for (; i < activities.length; i++) {
+    activities[i].data = data;
+    activities[i].activity = activity;
+    activities[i].duration = duration;
+  }
+}
+addActivity("23-7-2018", "Amazon Prime", 10);
+addActivity("23-7-2018", "Youtube", 30);
+addActivity("24-7-2018", "MTV", 30);
+addActivity("24-7-2018", "Youtube", 30);
+addActivity("26-7-2018", "Amazon Prime", 30);
+console.log(activities);
+
+var totalUsage = 0;
+var j = 0; // extra feature to calculate with how many activities the usageLimite is reached.
+const usageLimit = 120;
+function showStatus() {
+  const totalActivities = activities.length;
+  for (var i = 0; i < activities.length; i++) {
+    totalUsage += activities[i].duration;
+    if (j === 0) {
+      if (totalUsage > usageLimit || totalUsage === usageLimit) {
+        var limitOfActivity = i + 1;
+        console.log(
+          "You have reached your usage limit with first " +
+            limitOfActivity +
+            " activities of your activities list, no more smartphoning for you!"
+        );
+        j = j + 1;
+      }
+    }
+  }
+  return [totalActivities, totalUsage];
+}
+const showMyStatus = showStatus(activities);
```

Comment:
Yes, i did i+1 because i indicates the index position, not the serial no. fx: for 3rd activity i=2.
Yes, J is a counter to keep track of when the usage limit is reached. 
Thanks :)